mesh/mesh.hpp

```diff
@@ -818,33 +873,77 @@ class Mesh
+   int AddVertexAtMeanCenter(const int *vi, const int nverts, int dim = 3);
 
+   /// Adds a segment to the mesh given by 2 vertices @a v1 and @a v2.
    int AddSegment(int v1, int v2, int attr = 1);
+   /// Adds a segment to the mesh given by 2 vertices @a vi.
    int AddSegment(const int *vi, int attr = 1);
 
+   /// Adds a triangle to the mesh given by 3 vertices @a v1 through @a v3.
    int AddTriangle(int v1, int v2, int v3, int attr = 1);
+   /// Adds a triangle to the mesh given by 3 vertices @a vi.
    int AddTriangle(const int *vi, int attr = 1);
+   /// Adds a triangle to the mesh given by 3 vertices @a vi.
    int AddTri(const int *vi, int attr = 1) { return AddTriangle(vi, attr); }
 
+   /// Adds a quadrilateral to the mesh given by 4 vertices @a v1 through @a v4.
    int AddQuad(int v1, int v2, int v3, int v4, int attr = 1);
+   /// Adds a quadrilateral to the mesh given by 4 vertices @a vi.
    int AddQuad(const int *vi, int attr = 1);
 
+   /// Adds a tetrahedron to the mesh given by 4 vertices @a v1 through @a v4.
    int AddTet(int v1, int v2, int v3, int v4, int attr = 1);
+   /// Adds a tetrahedron to the mesh given by 4 vertices @a vi.
    int AddTet(const int *vi, int attr = 1);
 
+   /// Adds a wedge to the mesh given by 6 vertices @a v1 through @a v6.
    int AddWedge(int v1, int v2, int v3, int v4, int v5, int v6, int attr = 1);
+   /// Adds a wedge to the mesh given by 6 vertices @a vi.
    int AddWedge(const int *vi, int attr = 1);
 
+   /// Adds a pyramid to the mesh given by 5 vertices @a v1 through @a v5.
    int AddPyramid(int v1, int v2, int v3, int v4, int v5, int attr = 1);
+   /// Adds a pyramid to the mesh given by 5 vertices @a vi.
    int AddPyramid(const int *vi, int attr = 1);
 
+   /// Adds a hexahedron to the mesh given by 8 vertices @a v1 through @a v8.
    int AddHex(int v1, int v2, int v3, int v4, int v5, int v6, int v7, int v8,
               int attr = 1);
+   /// Adds a hexahedron to the mesh given by 8 vertices @a vi.
    int AddHex(const int *vi, int attr = 1);
+   /// Adds 6 tetrahedrons to the mesh by splitting a hexahedron given by
+   /// 8 vertices @a vi.
    void AddHexAsTets(const int *vi, int attr = 1);
+   /// Adds 2 wedges to the mesh by splitting a hexahedron given by
+   /// 8 vertices @a vi.
    void AddHexAsWedges(const int *vi, int attr = 1);
+   /// Adds 6 pyramids to the mesh by splitting a hexahedron given by
+   /// 8 vertices @a vi.
    void AddHexAsPyramids(const int *vi, int attr = 1);
 
+   /// Adds 24 tetrahedrons to the mesh by splitting a hexahedron.
+   /// @a vi are the 8 vertices of the hexahedron, @a hex_face_verts has the
+   /// map from the 4 vertices of each face of the hexahedron to the index
+   /// of the point created at the center of the face, and @a attr is the
+   /// attribute of the new elements. See @a Make3D24TetsFromHex for usage.
+   void AddHexAs24TetsWithPoints(int *vi,
+                                 std::map<std::array<int, 4>, int>
```

Comment:
I think it would be better (more efficient) to use `std::unordered_map` here instead of `std::map`.